src/utils/string.ts

```diff
@@ -0,0 +1,62 @@
+/**
+ * Trim the text by the given length.
+ *
+ * @param content - The raw text content.
+ * @param length - The length to render.
+ * @returns The trimed text.
+ */
+export function trimText(content: string, length = 100) {
+  if (!content) {
+    return content;
+  }
+
+  const _content = content.trim();
+
+  const contentArr = [..._content];
+  if (contentArr.length <= length) {
+    return _content;
+  }
+
+  // Handle sepcial characters
+  return `${contentArr.slice(0, length).join('')}...`;
+}
+
+/**
+ * Trim the text by the heading length and tailing length.
+ *
+ * @param content - The raw text content.
+ * @param head - The heading length to render.
+ * @param tail - The tailing length to render.
+ * @returns The trimed text.
+ */
+export function trimTextByHeadTail(
+  content: string,
+  head: number,
+  tail: number,
+) {
+  if (!content) {
+    return content;
+  }
```

Comment:
This can't be the case with the current types. Since this function is not expected to be used outside of this project, I don't think there is a need to do this.

Reply:
are you prefer to remove this checking?

Reply:
updated